wazp/callbacks/roi.py

```diff
@@ -268,37 +268,51 @@ def set_roi_color_in_table(
             return cond_format
```

Comment:
above in line 177 (I cannot comment on lines without changes):

`frame_step` becomes 0 if `(num_frames / 4) < 500` . Could this be a problem? 

We may have videos with less than 2000 frames right? (actually the limit is 2004 frames because it's wrapped in an `int`).

I like the slider option, but an alternative to avoid issues with steps may be to just show the user the total number of frames and let them pick one from 0 to the max number of frames. Maybe something like showing [ ] / 2500, where [ ] is an editable input text box, and by default [ ] is populated with the midpoint of the video. Then we can have buttons for +1, -1, +10, -10 in case that frame is not convenient to quickly switch but stay in the vicinity. 

If you feel this is a big change for v0, feel free to add as an enhancement issue

Reply:
actually I think this may be the case in this video? note the -2 in the slider

![image](https://user-images.githubusercontent.com/33267254/228558781-574c64d2-9163-477d-9f9d-afa780afd5f5.png)

Reply:
mmm I tried again and it showed 294k frames so not sure... 😬

Reply:
I was just checking on a new video and I found this:

when I define ROIs for a video with no ROIs defined, and then click 'Save all', the message doesn't update to show they were successfully saved (it stays with 'detected unsaved changes'). They do save correctly though.

![image](https://user-images.githubusercontent.com/33267254/228560196-4fdf6ce9-f4e2-4d99-827b-dc42a6bd1b43.png)

Reply:
> frame_step becomes 0 if (num_frames / 4) < 500 . Could this be a problem?
> We may have videos with less than 2000 frames right? (actually the limit is 2004 frames because it's wrapped in an int).

Good catch! I now do the rounding to nearest 1000 only if the frame step is  <1000. 
```python
# Divide the number of frames into 4 steps
frame_step = int(num_frames / 4)
# Round to the nearest 1000 if step is > 1000
if frame_step > 1000:
    frame_step = int(frame_step / 1000) * 1000
```
Therefore, if `num_frames` < 4004, no rounding will occur. I suspect most videos will be many thousands of frames, but it's good to be robust to edge cases.

Apart from that, under the frame I now display "Showing frame "n" of "N", making things more explicit. 

>I like the slider option, but an alternative to avoid issues with steps may be to just show the user the total number of frames and let them pick one from 0 to the max number of frames. Maybe something like showing [ ] / 2500, where [ ] is an editable input text box, and by default [ ] is populated with the midpoint of the video. Then we can have buttons for +1, -1, +10, -10 in case that frame is not convenient to quickly switch but stay in the vicinity.

I opened this as a separate issue #71

Reply:
> actually I think this may be the case in this video? note the -2 in the slider

That is in fact a separate problem, and it sent me down a rabbit hole yesterday. So if the video cannot be read (e.g. if not linked via alias), `OpenCV` still tries to calculate frames and fails. However, that error was not properly caught and you end up with negative numbers sometimes. This is further complicated by the frame caching locally, because sometimes even if the video itself cannot be read, the app still displays the locally cached frame (from a previous session).

Anyhow, to make things less confusing and much more explicit, I now throw an error when the video cannot be read and raise an alert in the app. This was not very straightforward, because catching `OpenCV` exceptions in Python does not work smoothly (despite an existing `cv2.error` object which is supposed to do this job). Therefore I had to come up with some indirect ways of figuring out when the video loading had gone wrong. I now include these checks in the `utils.get_num_frames` and `utils.extract_frame` functions directly.

Reply:
nice workaround! ✨